django/utils/text.py

```diff
@@ -12,6 +15,27 @@
 from django.utils.translation import gettext as _
 from django.utils.translation import gettext_lazy, pgettext
 
+# https://html.spec.whatwg.org/#void-elements
+SELF_CLOSING_TAGS = {
+    "area",
+    "base",
+    "br",
+    "col",
+    "embed",
+    "hr",
+    "img",
+    "input",
+    "link",
+    "meta",
+    "param",
+    "source",
+    "track",
+    "wbr",
+    # Deprecated tags
+    "frame",
+    "spacer",
+}
```

Comment:
I'm struggling to move this. With `VOID_ELEMENTS` in `django.utils.html` the tests crash with:

```
  File "C:\Users\smith\projects\django\django\utils\text.py", line 14, in <module>
    from django.utils.html import VOID_ELEMENTS
ImportError: cannot import name 'VOID_ELEMENTS' from partially initialized module 'django.utils.html' (most likely due to a circular import) (C:\Users\smith\projects\django\django\utils\html.py)
```

Any ideas?

Reply:
Hmm. Annoying that there's a circular import somewhere, especially as that's the logical place to put this. _(Will have to have a look, but not at laptop right now.)_

Reply:
Ah, ok. So it looks like there is an import of `django.utils.text.normalize_newlines()` in `django.utils.html` and we're wanting to import `django.utils.html.VOID_ELEMENTS` in `django.utils.text`. I suspect that we'll need to push down one of these imports...

**Option 1:** Move the `from django.utils.text import normalize_newlines` import into `django.utils.html.linebreaks()`.

**Option 2:** Add the following to `TruncateHTMLParser` to avoid the circular import (and re-importing multiple times in different methods):

```python
class TruncateHTMLParser(HTMLParser):
    ...

    @cached_property
    def void_elements(self):
        from django.utils.html import VOID_ELEMENTS

        return VOID_ELEMENTS

    ...
```

Reply:
Looking at moving and `VOID_ELEMENTS` again this morning. I think we should add a backward compatibility note. I know it's not documented, but someone is going to be using it!

I also noticed that this will mean `VOID_ELEMENTS` is in `django.utils.html` but `BOOLEAN_ATTRIBUTES` (which also seems undocumented to me) would be in `django.test.html`. That feels a little inconsistent to me. Maybe we should move both, but that makes me even more nervous re backward compatibility.

Reply:
I'd leave `BOOLEAN_ATTRIBUTES` in `django.test.html`, it's not worth moving.